kornia/geometry/transform/imgwarp.py

```diff
@@ -660,9 +660,9 @@ def get_shear_matrix2d(center: Tensor, sx: Optional[Tensor] = None, sy: Optional
     sx_tan = torch.tan(sx)
     sy_tan = torch.tan(sy)
     ones = torch.ones_like(sx)
-    shear_mat = stack([ones, -sx_tan, sx_tan * y, -sy_tan, ones + sx_tan * sy_tan, sy_tan * (sx_tan * y + x)], -1).view(
-        -1, 2, 3
-    )
+    shear_mat = stack(
+        [ones, -sx_tan, sx_tan * y, -sy_tan, ones + sx_tan * sy_tan, sy_tan * (x - sx_tan * y)], dim=-1
```

Comment:
if this math is incorrect -- can you add an additional unit test showing the fix? /cc @ducha-aiki

Reply:
@edgarriba Hi! I am not quite sure how to show that the bug is fixed with additional unit test - can you please give me a hint regarding that? Should I compare the 'old math' and 'new math' to see which one is the correct one? or what the best way would be to do this ? thanks:)

Reply:
so would it be ok if I test whether expected (controlled matrix with numbers) is the same as the matrix that is computed with changed get_shear_matrix2d function?

Reply:
that would work

Reply:
@edgarriba Added a unit test, let me know if it works :) thanks!